app/test/helpers/test_util.dart

```diff
@@ -0,0 +1,37 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+extension PumpUntilFound on WidgetTester {
```

Comment:
for reasons `pumpAndSettle` times out for us, so we have these helper functions, especially pumping the ProviderScope widget is super useful